### ISSUE TITLE
depends on outdated corestore

### ISSUE THREAD

**Issue by juliangruber** [p2pcommons/sdk-js#58]
The sdk depends on version `^2.0.0` of corestore, which isn't compatible with hyperdrive. This test file passes in the sdk repo but fails in the hypergraph repo, probably because npm resolves dependencies differently:

```js
const Commons = require('@p2pcommons/sdk-js')

;(async () => {
  const commons = new Commons({ baseDir: `/tmp/${Math.random()}` })
  await commons.ready()
  await commons.init({ type: 'content', title: 'title' })
  await commons.destroy()
})()
```

The error I get is this:

```
TypeError: this._corestore.ready is not a function
    at Hyperdrive._ready (/Users/julian/dev/hypergraph-xyz/cli/node_modules/hyperdrive/index.js:105:28)
    at run (/Users/julian/dev/hypergraph-xyz/cli/node_modules/thunky/index.js:19:5)
    at Hyperdrive.thunk [as ready] (/Users/julian/dev/hypergraph-xyz/cli/node_modules/thunky/index.js:13:5)
    at new Hyperdrive (/Users/julian/dev/hypergraph-xyz/cli/node_modules/hyperdrive/index.js:78:10)
    at module.exports.args (/Users/julian/dev/hypergraph-xyz/cli/node_modules/hyperdrive/index.js:28:31)
    at new HyperdrivePromise (/Users/julian/dev/hypergraph-xyz/cli/node_modules/@geut/hyperdrive-promise/index.js:12:16)
    at module.exports.args (/Users/julian/dev/hypergraph-xyz/cli/node_modules/@geut/hyperdrive-promise/index.js:54:31)
    at Object.create (/Users/julian/dev/hypergraph-xyz/cli/node_modules/@p2pcommons/sdk-js/lib/dat-helper.js:57:17)
    at SDK.init (/Users/julian/dev/hypergraph-xyz/cli/node_modules/@p2pcommons/sdk-js/index.js:292:36)
```

The corestore version installed locally, 2.0.0, doesn't have a `ready` method.

**Comment by juliangruber** [comment]
In fact I get two different dependency trees:

```bash
+ sdk-js (master) : npm ls corestore
@p2pcommons/sdk-js@0.3.2 /Users/julian/dev/p2pcommons/sdk-js
├─┬ @geut/hyperdrive-promise@1.0.0
│ └─┬ hyperdrive@10.3.2
│   └── corestore@2.0.0  deduped
└── corestore@2.0.0

+ cli (master) : npm ls corestore
@hypergraph-xyz/cli@0.2.14 /Users/julian/dev/hypergraph-xyz/cli
└─┬ @p2pcommons/sdk-js@0.3.2
  ├─┬ @geut/hyperdrive-promise@1.0.1
  │ └─┬ hyperdrive@10.4.4
  │   └── corestore@4.0.9
  └── corestore@2.0.0

```

**Comment by juliangruber** [comment]
This can be reproduced by `npm install "@geut/hyperdrive-promise@latest"` in this repo. It looks like there might have been a breaking change between 1.0.0 and 1.0.1

**Comment by juliangruber** [comment]
WIP: https://github.com/p2pcommons/sdk-js/compare/fix/corestore?expand=1

**Comment by dpaez** [comment]
related to: https://github.com/p2pcommons/sdk-js/issues/57

**Comment by dpaez** [comment]
Closed with #57
